File: frontend/lib/types/entry.ts
export interface Entry {
  id: number;
  user_id: number;
  image_url: string;
  image_hash: string;
  image_size: number;
  verified: boolean;
  admin_notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface EntryCreateResponse {
  success: boolean;
  entry: Entry;
  entries_remaining: number;
}

export interface DuplicateCheckResponse {
  is_duplicate: boolean;
  message: string;
  existing_entry_id?: number;
}

export interface EntryListResponse {
  success: boolean;
  entries: Entry[];
  total_entries: number;
  entries_remaining: number;
}

export interface CodeValidateRequest {
  code: string;
}

export interface CodeValidateResponse {
  success: boolean;
  message: string;
  code_id: number | null;
}

File: app/schemas/entry.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List, Dict

class EntryBase(BaseModel):
    pass

class EntryCreate(BaseModel):
    pass  # File upload handled separately

class EntryResponse(BaseModel):
    id: int
    user_id: int
    image_url: str
    verified: bool
    fraud_score: float
    review_status: str
    created_at: datetime

    class Config:
        from_attributes = True

class EntryDetailResponse(EntryResponse):
    fraud_reason: Optional[Dict] = None
    similar_entry_ids: Optional[List[int]] = None
    admin_notes: Optional[str] = None

class EntryListResponse(BaseModel):
    success: bool
    entries: list[EntryResponse]
    total_entries: int
    entries_remaining: int

class DuplicateCheckResponse(BaseModel):
    is_duplicate: bool
    message: str
